crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -393,14 +393,24 @@ def proposal_update
           if answer[0] == 202
             missing_nodes = answer[1].map { |node_dns| NodeObject.find_node_by_name(node_dns) }
 
-            unready_nodes = missing_nodes.map(&:alias)
+            unready_nodes = missing_nodes.select { |n| n.state != "ready" }.map(&:alias)
             unallocated_nodes = missing_nodes.reject(&:allocated?).map(&:alias)
 
-            flash[:notice] = "#{t('barclamp.proposal_show.commit_proposal_queued')}: #{(unready_nodes - unallocated_nodes).join(", ")}"
-
+            unless unready_nodes.empty?
+              flash[:notice] = "#{t('barclamp.proposal_show.commit_proposal_queued')}: #{(unready_nodes - unallocated_nodes).join(", ")}"
+            end
             unless unallocated_nodes.empty?
               flash[:alert]  = "#{t('barclamp.proposal_show.commit_proposal_queued_unallocated')}: #{unallocated_nodes.join(", ")}"
             end
+            if unready_nodes.empty? && unallocated_nodes.empty?
+              # find out which proposals were not applied yet
+              deps = @service_object.proposal_dependencies(ServiceObject.proposal_to_role(@proposal, params[:barclamp]))
+              missing_barclamps = deps.select.map do |dep|
+                depprop = Proposal.where(barclamp: dep["barclamp"], name: dep["inst"]).first
+                dep["barclamp"] unless depprop["deployment"][dep["barclamp"]]["crowbar-status"] == "success"
+              end.compact
```

Comment:
Btw, it might be good to actually rework things a bit so we can get this information directly from the deployment queue. This would involve not just returning the nodes that are not ready, but a structure with the deps that are not ready and the nodes that are not ready.